pages/docs/reference/lambdas.md

```diff
@@ -7,156 +7,220 @@ title: "Higher-Order Functions and Lambdas"
 
 # Higher-Order Functions and Lambdas
 
-## Higher-Order Functions
-
-A higher-order function is a function that takes functions as parameters, or returns a function.
-A good example of such a function is `lock()` that takes a lock object and a function, acquires the lock, runs the function and releases the lock:
+## Function types
+
+Kotlin functions are [*first-class*](https://en.wikipedia.org/wiki/First-class_function), which means that they can 
+be stored in variables and data structures, passed as arguments to and returned from other 
+[higher-order functions](#higher-order-functions), and operated with in any way that is possible for other 
+non-function values. 
+
+To facilitate this, Kotlin, as a statically typed programming language, uses a family of 
+types to represent functions and provides a set of specialized language constructs, such as [lambda expressions](#lambda-expressions-and-anonymous-functions). 
+
+The function types have special notation that corresponds to the signatures of the functions:
+
+* All function types have a parenthesized parameter types list and a return type: `(A, B) -> C` denotes a type that consists of 
```

Comment:
"consists of functions" - I'm not sure that it's the right terminology. "Represents functions"?